libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -1135,6 +1148,12 @@ void WaveTrack::ClearAndPaste(double t0, // Start of time to clear
    assert(IsLeader());
    assert(src.IsLeader());
    assert(srcNChannels == 1 || srcNChannels == NChannels());
+
+   // This has to be before `src.GetStartTime()` and such, or the boundaries
+   // won't be correct.
+   const auto finally =
+      ProvisionallySynchronizeProjectTempo(const_cast<WaveTrack&>(src));
```

Comment:
I still dislike the `const_cast` in two places and the non-obvious reasoning why there is never a net side effect on `WaveClip`.

An alternative is that `ProvisionallySynchronizeProjectTempo` (renamed maybe) takes a reference to a `const WaveTrack` and returns a shared pointer to a new, duplicate WaveTrack that has the project tempo change in it.  Its scope is only as long as this `finally` would be.

Maybe a lot of memory allocations and deallocations but a sharing of all the blocks in the Sqlite database so none of the very expensive things are done.

Reply:
Consider too that duplication of the in-memory WaveTrack structures, but with sharing of Sqlite resources, happens all the time when you push and pop the Undo stack, or even just modify the current entry because of a change of the selected interval.  And we hardly notice the performance.

Reply:
That's a good idea, better than this kind of magic I was doing.
See 8604d5726a239849bdd80636a3d948c31e076ccd.